packages/typespec-vscode/test/extension/common/common-steps.ts

```diff
@@ -0,0 +1,240 @@
+ * folderName: The text in the top input box is usually the current open root directory,
+ * command: After the top input box pops up, the command to be executed
+ */
+async function startWithCommandPalette(
+  page: Page,
+  { folderName, command }: { folderName: string; command: string },
+) {
+  await sleep(2);
+  await page.locator("li").filter({ hasText: folderName }).first().click();
+  await sleep(2);
+  await screenshot(page, "linux", "open_top_panel");
+  await page
+    .getByRole("textbox", { name: "Search files by name (append" })
+    .first()
+    .fill(`>TypeSpec: ${command}`);
+  let listForCreate: Locator;
+  await retry(
+    page,
+    5,
+    async () => {
+      listForCreate = page
+        .locator("a")
+        .filter({ hasText: `TypeSpec: ${command}` })
+        .first();
+      return (await listForCreate.count()) > 0;
+    },
+    "Failed to find the specified option",
+  );
+  await screenshot(page, "linux", "input_command");
+  await listForCreate!.click();
+}
+
+/**
+ * In vscode, when you need to select a folder or a file, call this method
+ * @param file When selecting a file, just pass it in. If you need to select a folder, you do not need to pass this parameter in.
+ */
+async function selectFolder(page: Page, file: string = "") {
+  await sleep(10);
+  await keyboard.type(file);
+  await screenshot(page, "linux", "select_folder");
+  await keyboard.pressKey(Key.Enter);
+  await keyboard.releaseKey(Key.Enter);
+  await sleep(3);
+}
+
+/**
+ * If the current folder is not empty, sometimes a pop-up will appear
+ * asking "Do you want to continue selecting the current folder as the root directory?".
+ * In this method, select "yes" because selecting "no" does not make sense.
+ * @param page vscode object
+ */
+async function notEmptyFolderContinue(page: Page) {
+  let yesBtn: Locator;
+  await retry(
+    page,
+    5,
+    async () => {
+      yesBtn = page
+        .getByRole("option", { name: "Yes" })
+        .locator("label")
+        .filter({ hasText: "Yes" })
+        .first();
+      const noBtn = page
+        .getByRole("option", { name: "No" })
+        .locator("label")
+        .filter({ hasText: "No" })
+        .first();
+      return (await yesBtn.count()) > 0 && (await noBtn.count()) > 0;
+    },
+    "Failed to find yes/no button",
+    1,
+  );
+  await retry(
+    page,
+    5,
+    async () => {
+      const yesdescriptionBox = page.getByRole("option", { name: "Yes" }).locator("label");
+      const yesdescriptionText = await yesdescriptionBox.textContent();
+      return yesdescriptionText !== null && yesdescriptionText.includes("YesSelected folder");
+    },
+    "Failed to match the description for the non-empty folder cases",
+    1,
+  );
+  await screenshot(page, "linux", "not_empty_folder_continue");
+  await yesBtn!.click();
+}
+
+/**
+ * Install plugins using the command
+ */
+async function installExtensionForCommand(page: Page, extensionDir: string) {
```

Comment:
Back to the previous comment I think this can be greatly simplified by using the command line to install the extension. This feels like a lot of code that is 100% irrelevant to testing our extensions

See how `tsp code insall` cli command works to install the vsix via the command line https://github.com/microsoft/typespec/blob/main/packages/compiler/src/core/cli/install-vsix.ts